### ISSUE TITLE
v0.7.3 Fails to establish client connection with Hasura

### ISSUE THREAD

**Issue by pankgeorg** [DeloitteDigitalAPAC/GraphQLClient.jl#14]
```julia
pgeorgakopoulos@rosetta:~/julia/JuliaTeam$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> import Pkg

julia> Pkg.status()
      Status `~/.julia/environments/v1.7/Project.toml`
  [09d831e3] GraphQLClient v0.7.2
  [c3e4b0f8] Pluto v0.18.1 `../../../pluto/Pluto.jl`

julia> import GraphQLClient
julia> GraphQLClient.global_graphql_client(GraphQLClient.Client("http://localhost:9001/v1/graphql"))
GraphQLClient Client
       endpoint: http://localhost:9001/v1/graphql
    ws_endpoint: ws://localhost:9001/v1/graphql

julia> 
pgeorgakopoulos@rosetta:~/julia/JuliaTeam$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.7) pkg> upgrade
ERROR: `upgrade` is not a recognized command. Type ? for help with available commands

(@v1.7) pkg> update
    Updating registry at `~/.julia/registries/General.toml`
   Installed FuzzyCompletions ─ v0.5.0
    Updating `~/.julia/environments/v1.7/Project.toml`
  [09d831e3] ↑ GraphQLClient v0.7.2 ⇒ v0.7.3
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [5218b696] ↑ Configurations v0.17.0 ⇒ v0.17.3
  [fb4132e2] ↑ FuzzyCompletions v0.4.3 ⇒ v0.5.0
  [09d831e3] ↑ GraphQLClient v0.7.2 ⇒ v0.7.3
  [0ae10fbf] + GraphQLParser v0.1.2
  [83e8ac13] ↑ IniFile v0.5.0 ⇒ v0.5.1
Precompiling project...
  6 dependencies successfully precompiled in 11 seconds (22 already precompiled)

julia> 

julia> import GraphQLClient

julia> GraphQLClient.global_graphql_client(GraphQLClient.Client("http://localhost:9001/v1/graphql"))
ERROR: MethodError: no method matching getindex(::Nothing, ::String)
Stacktrace:
 [1] full_introspection!(client::GraphQLClient.Client)
   @ GraphQLClient ~/.julia/packages/GraphQLClient/S5eor/src/introspection.jl:112
 [2] GraphQLClient.Client(endpoint::String, ws_endpoint::String; headers::Dict{Any, Any}, introspect::Bool)
   @ GraphQLClient ~/.julia/packages/GraphQLClient/S5eor/src/client.jl:78
 [3] Client
   @ ~/.julia/packages/GraphQLClient/S5eor/src/client.jl:62 [inlined]
 [4] #Client#1
   @ ~/.julia/packages/GraphQLClient/S5eor/src/client.jl:60 [inlined]
 [5] GraphQLClient.Client(endpoint::String)
   @ GraphQLClient ~/.julia/packages/GraphQLClient/S5eor/src/client.jl:60
 [6] top-level scope
   @ REPL[4]:1

julia> 
```

The error seems to be happening while introspecting: https://github.com/DeloitteDigitalAPAC/GraphQLClient.jl/blob/main/src/introspection.jl#L111-L112

Hasura logs also show an error, that `" is not valid GraphQL name"`, not sure what it means
```
{"type":"http-log","timestamp":"2022-02-25T10:20:16.772+0000","level":"error","detail":{"operation":{"error":{"path":"$.operationName","error":" is not valid GraphQL name","code":"parse-failed"},"request_id":"1514d9e1-13ba-4cc0-8dba-d3486a1953c4","response_size":85,"request_mode":"error"},"request_id":"1514d9e1-13ba-4cc0-8dba-d3486a1953c4","http_info":{"status":200,"http_version":"HTTP/1.1","url":"/v1/graphql","ip":"172.18.0.1","method":"POST","content_encoding":null}}}
```
Probably something in the introspection query is incompatible with the way hasura does it.

**Comment by mmiller-max** [comment]
Thanks for reporting this. I think this is the key bit from the error:

```
{"path":"$.operationName","error":" is not valid GraphQL name","code":"parse-failed"}
```
`operationName` refers to `IntrospectionQuery` ([here](https://github.com/DeloitteDigitalAPAC/GraphQLClient.jl/blob/3157bb887109887d5f6edd24d0369671969be827/src/introspection.jl#L11)).

Is it possible that you do not have access to the introspection, because of [this setting](https://hasura.io/docs/latest/graphql/cloud/security/disable-graphql-introspection.html)?

Alternatively, it may have a different name for the introspection query, but I think `IntrospectionQuery` is the default.

Either way, the error you actually saw is not very helpful. I think we need to pass `throw_on_execution_error=True` to [this](https://github.com/DeloitteDigitalAPAC/GraphQLClient.jl/blob/3157bb887109887d5f6edd24d0369671969be827/src/introspection.jl#L111) use of `execute` so that you actually see the error when introspection fails.

**Comment by pankgeorg** [comment]
We also need to be able to use the client without introspection; we want to allow anonymous access to GraphQL

**Comment by mmiller-max** [comment]
> We also need to be able to use the client without introspection; we want to allow anonymous access to GraphQL

Set the `introspect` kwarg to `false`:

```Julia
GraphQLClient.Client("http://localhost:9001/v1/graphql", introspect=false)
```

**Comment by pankgeorg** [comment]
I think the real issue here is the operationName being empty.

`GraphQLClient@0.7.2` sends this query to hasura:
```json
{"query": { "variables": {}, "query": "query Test {jobs{jobid}}" },}
```

`GraphQLClient@0.7.4` sends 
```
{"raw_query":"{\"variables\":{},\"operationName\":\"\",\"query\":\"query TEST {jobs{jobid}}\"}"}},
```

The operation name is wrong (empty instead of TEST) and that makes hasura crash.
The same is true for the introspection query - it works fine in 0.7.2; introspection is not an issue; it fails due to missing `operationName`

**Comment by mmiller-max** [comment]
Closed by #15
